### ISSUE TITLE
overfitting

### ISSUE THREAD

**Issue by YangXS** [cesc-park/CRCN#11]
I find that the validation loss will increase after only 3 stages.  Does the model have heavy over-fitting problem ?

Train on 5823 samples, validate on 648 samples
Epoch 0
5823/5823 [==============================] - 322s - loss: 14098.2655 - val. loss: 197782.5469
Epoch 1
5823/5823 [==============================] - 322s - loss: 5548.3110 - val. loss: 143680.6094
Epoch 2
5823/5823 [==============================] - 321s - loss: 3508.6181 - val. loss: 119612.8281
Epoch 3
5823/5823 [==============================] - 320s - loss: 2460.4376 - val. loss: 108164.6172
Epoch 4
5823/5823 [==============================] - 322s - loss: 1882.4434 - val. loss: 103302.0938
Checkpoint saved
.
.
.
Number of stage 3
Train on 5823 samples, validate on 648 samples
Epoch 0
5823/5823 [==============================] - 322s - loss: 678.7651 - val. loss: 91479.6172
Epoch 1
5823/5823 [==============================] - 321s - loss: 619.2658 - val. loss: 95824.9844
Epoch 2
5823/5823 [==============================] - 321s - loss: 572.6705 - val. loss: 102938.8516
Epoch 3
5823/5823 [==============================] - 322s - loss: 525.9343 - val. loss: 104955.4688
Epoch 4
5823/5823 [==============================] - 321s - loss: 481.5498 - val. loss: 102355.6406
Checkpoint saved
Number of stage 4
Train on 5823 samples, validate on 648 samples
Epoch 0
5823/5823 [==============================] - 322s - loss: 441.9791 - val. loss: 102219.4531
Epoch 1
5823/5823 [==============================] - 321s - loss: 430.3209 - val. loss: 111248.1094
Epoch 2
5823/5823 [==============================] - 320s - loss: 400.2233 - val. loss: 109078.7812
Epoch 3
5823/5823 [==============================] - 322s - loss: 393.7606 - val. loss: 103563.5000
Epoch 4
5823/5823 [==============================] - 321s - loss: 382.1619 - val. loss: 108578.5625
Checkpoint saved
.
.
.
Number of stage 19
Train on 5823 samples, validate on 648 samples
Epoch 0
5823/5823 [==============================] - 394s - loss: 231.7763 - val. loss: 252069.9531
Epoch 1
5823/5823 [==============================] - 388s - loss: 229.1790 - val. loss: 280088.4375
Epoch 2
5823/5823 [==============================] - 399s - loss: 231.8644 - val. loss: 245616.1719
Epoch 3
5823/5823 [==============================] - 412s - loss: 233.6944 - val. loss: 222301.6875
Epoch 4
5823/5823 [==============================] - 398s - loss: 228.1494 - val. loss: 291693.0938
Checkpoint saved

**Comment by cesc-park** [comment]
3 stages mean 15 epochs, that's enough I think.
It would be nice to test it by reducing the size of the model.

**Comment by YangXS** [comment]
Thank you for your reply.
I tested the model trained with 3 stages.
The evaluated results are still not good:
Bleu_1: 0.127
Bleu_2: 0.032
Bleu_3: 0.007
Bleu_4: 0.002

Could you give more details about how to reduce the size of the model ?

**Comment by cesc-park** [comment]
Did you set the same hyper-parameter settings as in the paper?

**Comment by cesc-park** [comment]
The code is for demonstrating examples. To make it work faster, there are smaller hyper-parameter values than the numbers in the paper. ex) TOPK = 3

**Comment by YangXS** [comment]
I actually only used the default hyper-parameter setting in the training and generation code.
I will try to fix that.
Thanks.
